interfaces/compatibility/compatibility.go

```diff
@@ -227,9 +150,25 @@ func checkCompatAgainstSpec(compatField *CompatField, spec *CompatSpec) error {
 	return nil
 }
 
-// checkCompatibility checks if two compatibility fields are compatible by
-// looking at the tags and at the intersection of the defined integer ranges.
-func CheckCompatibility(compat1, compat2 CompatField) bool {
+// CheckCompatibility checks the compatibility of two compatibility expressions.
+func CheckCompatibility(compat1, compat2 string) bool {
+	expr1, labels1, err := parse(compat1)
+	if err != nil {
+		return false
+	}
+	expr2, labels2, err := parse(compat2)
+	if err != nil {
+		return false
+	}
+
+	// Check cross-compatibility
```

Comment:
is not immediately obvious why it's ok to ignore the structuring of one side at a time, when doing the cross-checks. Can you have a comment here explaining this?

Reply:
Added